tests/test_idle.py

```diff
@@ -0,0 +1,42 @@
+#    Copyright 2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+
+import utils_tests
+import trappy
+
+sys.path.append(os.path.join(utils_tests.TESTS_DIRECTORY, "..", "trappy"))
+
+class BaseTestIdle(utils_tests.SetupDirectory):
+    def __init__(self, *args, **kwargs):
+        super(BaseTestIdle, self).__init__(
+             [("trace_idle.txt", "trace.txt"),
+              ("trace_idle.txt", "trace.raw.txt")],
+             *args,
+             **kwargs)
+
+class TestCpuIdle(BaseTestIdle):
+    def test_get_dataframe(self):
+        df = trappy.FTrace().cpu_idle.data_frame
+
+        exp_states = [2, -1, -1, 1, -1, 0, -1, 2, -1, 0, 0, -1, 0, -1, -1, -1, 2]
+        exp_cpus =   [5,  2,  1, 2,  5, 5,  3, 3,  0, 1, 0,  0, 0 , 0 , 5,  1, 5]
+
+        self.assertEqual(len(df), len(exp_states))
+
+        self.assertTrue((df["state"] == exp_states).all())
```

Comment:
If/When this fails, it gives an unhelpful message.  It's better to let unittest do the comparison, something like (untested):

``` python
self.assertEquals(df["state"].tolist(), exp_states)
```

Same for the `cpu_id` column.